src/apps/advanced-search/AdvancedSearchRow.tsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import IconExpand from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/ExpandMore.svg";
+import IconMinus from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/MinusButton.svg";
+import IconPlus from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/PlusButton.svg";
+import clsx from "clsx";
+import {
+  AdvancedSearchClause,
+  AdvancedSearchClauses,
+  AdvancedSearchIndex,
+  AdvancedSearchIndexTranslations,
+  AdvancedSearchIndexes,
+  AdvancedSearchQuery,
+  initialAdvancedSearchQuery,
+  AdvancedSearchRowData
+} from "./types";
+import { useText } from "../../core/utils/text";
+
+export type AdvancedSearchRowProps = {
+  data: AdvancedSearchQuery;
+  dataCy?: string;
+  rowIndex: number;
+  setSearchObject: (searchObject: AdvancedSearchQuery) => void;
+};
+
+const AdvancedSearchRow: React.FC<AdvancedSearchRowProps> = ({
+  dataCy = "advanced-search-row",
+  data,
+  rowIndex,
+  setSearchObject
+}) => {
+  const t = useText();
+
+  const updateRowData = (
+    rowAspect: "term" | "clause" | "searchIndex",
+    update: string | AdvancedSearchClause | AdvancedSearchIndex,
+    updateData: (data: AdvancedSearchQuery) => void
+  ) => {
+    const newData = { ...data };
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore-next-line
+    newData.rows[rowIndex][rowAspect] = update;
+    updateData(newData);
+  };
+
+  const getClauseClasses = (
+    clickedClause: AdvancedSearchClause["value"],
+    currentClause: AdvancedSearchClause["value"]
+  ) => {
+    return clsx(
+      "advanced-search__clause focus-styling cursor-pointer capitalize-all",
+      {
+        "advanced-search__clause--grey": currentClause !== clickedClause
```

Comment:
In this case you, potentially can end up with both classes:
"advanced-search__clause" and "advanced-search__clause--grey"
right? Or is that not possible?

Reply:
that's absolutely correct! And that's also the wat it's supposed to be as `advanced-search__clause--grey` is a modifier class :) and for those you need the base Block class too present.